RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -61,40 +61,130 @@ public bool Generate(ProcessedArgs args)
+                this.logger.LogWarning(UIResources.APG_NoAnalyzersFound, targetPackage.Id);
 
-            IEnumerable<IPackage> licenseAcceptancePackages = this.GetPackagesRequiringLicenseAcceptance(package);
-            if (licenseAcceptancePackages.Any() && ! args.AcceptLicenses)
+                if (!args.RecurseDependencies || !dependencyPackages.Any())
+                {
+                    this.logger.LogWarning(UIResources.APG_NoAnalyzersInTargetSuggestRecurse);
+                    return false;
+                }
+            }
+
+            if (args.RecurseDependencies)
             {
-                this.logger.LogError(UIResources.APG_NGPackageRequiresLicenseAcceptance, package.Id, package.Version.ToString());
+                // Possible sub-case - target package has dependencies that contain analyzers
+                foreach (IPackage dependencyPackage in dependencyPackages)
+                {
+
+                    if (HasAnalyzers(dependencyPackage, args.Language))
+                    {
+                        packagesContainingAnalyzers.Add(dependencyPackage);
+                    }
+                    else
+                    { 
+                        this.logger.LogWarning(UIResources.APG_NoAnalyzersFound, dependencyPackage.Id);
+                    }
+                }
+
+                if (!packagesContainingAnalyzers.Any())
+                {
+                    return false;
+                }
+            }
+
+            // Check for packages that require the user to accept a license
+            IEnumerable<IPackage> licenseAcceptancePackages = this.GetPackagesRequiringLicenseAcceptance(targetPackage);
+            if (licenseAcceptancePackages.Any() && !args.AcceptLicenses)
+            {
+                // NB: This warns for all packages under the target that require license acceptance
+                // (even if they aren't related to the packages from which plugins were generated)
+                this.logger.LogError(UIResources.APG_NGPackageRequiresLicenseAcceptance, targetPackage.Id, targetPackage.Version.ToString());
                 this.ListPackagesRequiringLicenseAcceptance(licenseAcceptancePackages);
                 return false;
             }
 
+            IEnumerable<IPackage> dependencyPackagesContainingAnalyzers = packagesContainingAnalyzers;
+            List<string> generatedJarFiles = new List<string>();
+            // Initial run with the user-targeted package and arguments
+            if (packagesContainingAnalyzers.Contains(targetPackage))
+            {
+                string generatedJarPath = GeneratePluginForPackage(targetPackage, args.Language, args.OutputDirectory, args.SqaleFilePath);
+                if (generatedJarPath == null)
+                {
+                    return false;
+                }
+
+                generatedJarFiles.Add(generatedJarPath);
+                dependencyPackagesContainingAnalyzers = dependencyPackagesContainingAnalyzers.Skip(1);
```

Comment:
_List.Contains_ will return _true_ if the item appears anywhere in the list, so you should use _List.Remove_ to remove it.

If you're going to use _Skip(1)_ then you should explicitly check that the target package is the first item in the list e.g. _List.FirstOrDefault_.